### PR TITLE
Add hosting consumer RPM at /pub over HTTPS

### PR DESCRIPTION
**Pull request by ehelms**
Adds a container that builds a consumer RPM and deploys it to `/var/www/html/pub` which is then served up by the content-server. To register a client:

```
wget --no-check-certificate https://<hostname>/pub/katello-rhsm-consumer-1.0-1.noarch.rpm
yum install katello-rhsm-consumer-1.0-1.noarch.rpm
FOREMAN_HOSTNAME=<hostname> /usr/bin/katello-rhsm-consumer
```

This workflow varies a bit from a traditional install and can be cleaned up over time. The difficulty with the standard method is that the hostname is embedded in the consumer RPM which is difficult to determine in a minishift type setup where the hostname is more dynamic. The RPM also has to be hosted over HTTPS for now until further digging into allowing HTTPS and HTTP through a single hostname is investigated.

### CODE REVIEW

File: containers/roles/client-rpm/files/build-client-rpm/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+- name: 'Make build directory'
+  file:
+    state: directory
+    path: "/root/build"
+
+- name: 'Read ca.crt'
+  set_fact:
+    ca_crt: "{{ lookup('file', '/etc/pki/katello/certs/katello-default-ca.crt') }}"
```
Comment (carbonin) [inline containers/roles/client-rpm/files/build-client-rpm/tasks/main.yml:9]: Does this need to be added at build time? Is this something that will ever need to be replaced by a user? They would have to rebuild the container right?

**Comment by carbonin** [comment]
It looks like this adds a new PVC. What kind of data is on there and is that something else we need to import for the upgrade?

File: containers/roles/client-rpm/files/build-client-rpm/tasks/main.yml (hunk 1 above)
Comment (ehelms) [reply containers/roles/client-rpm/files/build-client-rpm/tasks/main.yml:9]: This container is a bit of Ansible-inception so bear with me. The built image contains a role that performs this task. Then when the container is executed it looks up this data from the mounted secret, generates an RPM from it and pushes that RPM to the pub/ PVC. The only time this needs to be re-run is if the CA crt is regenerated which is a super rare event. I don't think overall this is the cleanest solution but I am attempting to mimic the behavior of existing installs. I do wonder if the Openshift part should be a Job rather than a deployment? I am less familiar with Jobs but saw you use them for the data import.

Comment (carbonin) [reply containers/roles/client-rpm/files/build-client-rpm/tasks/main.yml:9]: Ah this file (`main.yml`) is being built into the image itself.

A Job is a "run a container until success" sort of operation, so if this does not need to exist after running and will only run once, then a Job sounds like a good fit.

**Comment by ehelms** [comment]
The data stored there is:

 a) consumer RPM for registration
 b) CA certificate so users can easily download it to import and trust it
 c) bootstrap.py which is a script that users can use to automate some client actions

The bootstrap.py script comes from install katello-client-bootstrap RPM. In a true environment clone, we'd want to grab the consumer RPM and CA certificate. However, the consumer RPM in this case stores the hostname inside the RPM. So it will work if the hostname remains the same and that itself becomes an interesting exercise.

TLDR; Yes, we want to import it over, but it may present some challenges with the data thats actually present.

File: containers/roles/client-rpm/files/build-client-rpm/tasks/main.yml (hunk 1 above)
Comment (ehelms) [reply containers/roles/client-rpm/files/build-client-rpm/tasks/main.yml:9]: Updated to Job type and moved into a jobs file

**Review by carbonin — APPROVED** [review]
